### PR TITLE
feature: allow Automatic IAM Authentication without a password

### PR DESCRIPTION
**Pull request by LukeSchlangen**

### CODE REVIEW

File: cloudsql/postgres/database-sql/connect_connector.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022 Google LLC
+// Copyright 2023 Google LLC
```
Comment (jackwotherspoon) [inline cloudsql/postgres/database-sql/connect_connector.go:1]: nit: License headers should aim to reflect the year the file was originally created, updating the file does not require the license header to be changed.
```suggestion
// Copyright 2022 Google LLC
```

File: cloudsql/postgres/database-sql/connect_tcp.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022 Google LLC
+// Copyright 2023 Google LLC
```
Comment (jackwotherspoon) [inline cloudsql/postgres/database-sql/connect_tcp.go:1]:
```suggestion
// Copyright 2022 Google LLC
```

File: cloudsql/postgres/database-sql/connect_unix.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022 Google LLC
+// Copyright 2023 Google LLC
```
Comment (jackwotherspoon) [inline cloudsql/postgres/database-sql/connect_unix.go:1]:
```suggestion
// Copyright 2022 Google LLC
```

File: cloudsql/postgres/database-sql/deployment.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2021 Google LLC
+# Copyright 2023 Google LLC
```
Comment (jackwotherspoon) [inline cloudsql/postgres/database-sql/deployment.yaml:1]:
```suggestion
# Copyright 2021 Google LLC

```

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]
We recently split out the Postgres auto IAM AuthN sample into its own file [connect_connector_iam_authn.go](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/cloudsql/postgres/database-sql/connect_connector_iam_authn.go) as part of #2836. The aim was to try and make the connection path for auto IAM AuthN more clear to users within our samples and docs.  It does not require a password and I believe makes this PR a duplicate potentially. I still think your error and comment improvements are great adds and worth commiting.

I'll let @enocom weigh in as he may like this implementation better than the current one 😄

**Comment by enocom** [comment]
@LukeSchlangen Can we make the error and comment improvements on the split out sample?

File: cloudsql/postgres/database-sql/connect_connector.go (hunk 1 above)
Comment (LukeSchlangen) [reply cloudsql/postgres/database-sql/connect_connector.go:1]: Happy to switch these back. I had a different PR rejected for not having the current year. Is this a repository-by-repository choice? https://github.com/GoogleCloudPlatform/cloud-sql-proxy/runs/10259552138

Comment (enocom) [reply cloudsql/postgres/database-sql/connect_connector.go:1]: If the file already exists, then the older year takes precedence. If it's a new file, then naturally the current year is the way to go.

Comment (LukeSchlangen) [reply cloudsql/postgres/database-sql/connect_connector.go:1]: Perfect. Makes total sense. Thank you!

**Comment by LukeSchlangen** [comment]
@enocom Yes! Good idea. I set up a meeting with @jackwotherspoon for Wednesday to figure out the best way to merge my changes with his. I'll update this PR after that.

**Comment by LukeSchlangen** [comment]
Closing in favor of #2862

**Pull request closed without merging**